gui/src/3D/gui3DManager.ts

```diff
@@ -203,6 +204,17 @@ export class GUI3DManager implements IDisposable {
 
         this._sharedMaterials = {};
 
+        for (var materialName in this._touchSharedMaterials) {
+            if (!this._touchSharedMaterials.hasOwnProperty(materialName)) {
+                continue;

```

Comment:
nit: are we expecting this collection to change as we iterate over it?

Reply:
I am not sure, you think we might need a lock of some kind?

Reply:
no, i just don't think we need to check whether the materialName is in the collection when we are iterating over the collection :)

---
In reply to: [545508015](https://github.com/BabylonJS/Babylon.js/pull/9620#discussion_r545508015) [](ancestors = 545508015)

Reply:
That's a good point... 
@deltakosh As the one who added this check back in 2018 to this._sharedMaterials, do you happen to remember what the check was for?